plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSystemTables.java

```diff
@@ -99,4 +108,353 @@ public void testPropertiesTable()
+                """);
+        testPartitionsTableDataColumn("BIGINT", "_bigint", "VALUES (3, 'a'), (6, 'a'), (0, 'b'), (9, 'b')", 4, """
+                VALUES
+                ROW(ROW(BIGINT '3', BIGINT '6', BIGINT '0')),
+                ROW(ROW(BIGINT '0', BIGINT '9', BIGINT '0'))
+                """);
+        testPartitionsTableDataColumn("REAL", "_real", "VALUES (10.3, 'a'), (15.7, 'a'), (3.2, 'b'), (6.1, 'b')", 4, """
+                VALUES
+                ROW(ROW(REAL '10.3', REAL '15.7', BIGINT '0')),
+                ROW(ROW(REAL '3.2', REAL '6.1', BIGINT '0'))
+                """);
+        testPartitionsTableDataColumn("DOUBLE", "_double", "VALUES (3.2, 'a'), (7.25, 'a'), (7.25, 'b'), (18.9382, 'b')", 4, """
+                VALUES
+                ROW(ROW(DOUBLE '3.2', DOUBLE '7.25', BIGINT '0')),
+                ROW(ROW(DOUBLE '7.25', DOUBLE '18.9382', BIGINT '0'))
+                """);
+        testPartitionsTableDataColumn("DECIMAL(10)", "_decimal", "VALUES (5.6, 'a'), (1.2, 'a'), (532.62, 'b'), (153.27, 'b')", 4, """
+                VALUES
+                ROW(ROW(CAST(1.2 AS DECIMAL(10)), CAST(5.6 AS DECIMAL(10)), BIGINT '0')),
+                ROW(ROW(CAST(153.27 AS DECIMAL(10)), CAST(532.62 AS DECIMAL(10)), BIGINT '0'))
+                """);
+        testPartitionsTableDataColumn("DATE", "_date", "VALUES (CAST('2019-09-08' AS DATE), 'a'), (CAST('2020-09-08' AS DATE), 'a'), (CAST('2019-09-07' AS DATE), 'b'), (CAST('2019-09-08' AS DATE), 'b')", 4, """
+                VALUES
+                ROW(ROW(DATE '2019-09-08', DATE '2020-09-08', BIGINT '0')),
+                ROW(ROW(DATE '2019-09-07', DATE '2019-09-08', BIGINT '0'))
+                """);
+        testPartitionsTableDataColumn("TIMESTAMP(6)", "_timestamp", "VALUES (TIMESTAMP '2001-05-06 12:34:56.123456', 'a'), ( TIMESTAMP '2001-05-06 12:34:56.567890', 'a'), (TIMESTAMP '2001-05-06 12:34:56.123456', 'b'), ( TIMESTAMP '2001-05-06 12:34:56.123457', 'b')", 4, """
+                VALUES
+                ROW(ROW(CAST(NULL AS TIMESTAMP(6)), CAST(NULL AS TIMESTAMP(6)), BIGINT '0')),
+                ROW(ROW(CAST(NULL AS TIMESTAMP(6)), CAST(NULL AS TIMESTAMP(6)), BIGINT '0'))
+                """);
+        testPartitionsTableDataColumn("VARCHAR", "_varchar", "VALUES ('z', 'a'), ('x', 'a'), ('a', 'b'), ('b', 'b')", 4, """
+                VALUES
+                ROW(ROW(CAST('x' AS VARCHAR), CAST('z' AS VARCHAR), BIGINT '0')),
+                ROW(ROW(CAST('a' AS VARCHAR), CAST('b' AS VARCHAR), BIGINT '0'))
+                """);
+        testPartitionsTableDataColumn("VARBINARY", "_varbinary", "VALUES (VARBINARY 'abcd', 'a'), (VARBINARY 'jkl', 'a'), (VARBINARY 'mno', 'b'), (VARBINARY 'xyzz', 'b')", 4, """
+                VALUES
+                ROW(ROW(CAST(NULL AS VARBINARY), CAST(NULL AS VARBINARY), BIGINT '0')),
+                ROW(ROW(CAST(NULL AS VARBINARY), CAST(NULL AS VARBINARY), BIGINT '0'))
+                """);
```

Comment:
Please add the following test types:
* Longer decimal type (p > 18)
* TIMESTAMP(3) WITH TIME ZONE
* Array
* Map
* Row